great_expectations/expectations/self_check_util.py

```diff
@@ -617,222 +619,17 @@ def get_dataset(
-                        sparktypes.StructField(column, sparktypes.StringType())
-                        for column in schema
-                    ]
-                )
-                spark_df = spark.createDataFrame(data_reshaped, string_schema)
-                for c in spark_df.columns:
-                    spark_df = spark_df.withColumn(
-                        c, spark_df[c].cast(SPARK_TYPES[schema[c]]())
-                    )
-        elif len(data_reshaped) == 0:
-            # if we have an empty dataset and no schema, need to assign an arbitrary type
-            columns = list(data.keys())
-            spark_schema = sparktypes.StructType(
-                [
-                    sparktypes.StructField(column, sparktypes.StringType())
-                    for column in columns
-                ]
-            )
-            spark_df = spark.createDataFrame(data_reshaped, spark_schema)
-        else:
-            # if no schema provided, uses Spark's schema inference
-            columns = list(data.keys())
-            spark_df = spark.createDataFrame(data_reshaped, columns)
-
-        if table_name is None:
-            table_name = "test_data_" + "".join(
-                [random.choice(string.ascii_letters + string.digits) for _ in range(8)]
-            )
-
-        return _build_spark_validator_with_data(df=spark_df)
-
-    else:
-        raise ValueError("Unknown dataset_type " + str(execution_engine))
-
-
-def _build_spark_engine(df):
-    from pyspark.sql import SparkSession
-
-    spark = SparkSession.builder.getOrCreate()
-    if isinstance(df, pd.DataFrame):
-        df = spark.createDataFrame(
-            [
-                tuple(
-                    None if isinstance(x, (float, int)) and np.isnan(x) else x
-                    for x in record.tolist()
-                )
-                for record in df.to_records(index=False)
-            ],
-            df.columns.tolist(),
-        )
-    batch = Batch(data=df)
-    engine = SparkDFExecutionEngine(batch_data_dict={batch.id: batch.data})
-    return engine
-
-
-def _build_spark_validator_with_data(df):
-    from pyspark.sql import SparkSession
-
-    spark = SparkSession.builder.getOrCreate()
-    if isinstance(df, pd.DataFrame):
-        df = spark.createDataFrame(
-            [
-                tuple(
-                    None if isinstance(x, (float, int)) and np.isnan(x) else x
-                    for x in record.tolist()
-                )
-                for record in df.to_records(index=False)
-            ],
-            df.columns.tolist(),
-        )
-    batch = Batch(data=df)
-
-    return Validator(execution_engine=SparkDFExecutionEngine(), batches=(batch,))
-
-
 def _build_sa_engine(df):
     import sqlalchemy as sa
 
-    eng = sa.create_engine("sqlite://", echo=False)
-    df.to_sql("test", eng)
-    batch_data = SqlAlchemyBatchData(engine=eng, table_name="test")
+    execution_engine = sa.create_engine("sqlite://", echo=False)
```

Comment:
nit -- let's leave this "engine" or "sqlalchemy_engine" since it's not a GE execution engine

Reply:
@jcampbell : Done -- please re-review.  Thank you.